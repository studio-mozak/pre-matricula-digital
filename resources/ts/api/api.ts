import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
