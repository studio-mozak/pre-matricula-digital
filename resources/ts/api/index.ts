export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
