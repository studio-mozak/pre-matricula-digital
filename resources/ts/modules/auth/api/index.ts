/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
