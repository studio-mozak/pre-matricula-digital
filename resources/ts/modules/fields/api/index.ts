/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
