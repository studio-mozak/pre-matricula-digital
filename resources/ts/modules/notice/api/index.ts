/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
