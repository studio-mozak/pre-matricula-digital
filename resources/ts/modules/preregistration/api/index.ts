/**
 * Exporting the Axios Instance
 */
export { http, graphql, rest } from '@/api';

/**
 * Exporting all modules
 */
export * as Preregistration from './services/graphql/preregistration';
export * as PreregistrationRest from './services/rest/preregistration';
