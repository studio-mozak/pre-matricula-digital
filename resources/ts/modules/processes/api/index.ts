/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Process from './services/graphql/process';
export * as ProcessField from './services/graphql/processField';
export * as ProcessGrouper from './services/graphql/processGrouper';
