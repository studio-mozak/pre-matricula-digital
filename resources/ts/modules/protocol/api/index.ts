/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Protocol from './services/graphql/protocol';
