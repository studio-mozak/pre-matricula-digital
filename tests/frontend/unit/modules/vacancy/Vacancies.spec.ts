import '@/plugin/sortby';
import {
  VacancyFilter,
  VacancyLoadProcesses,
  VacancyTotalProcess,
} from '@/modules/vacancy/types';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import VTooltip from 'v-tooltip';
import Vacancies from '@/modules/vacancy/components/Vacancies.vue';
import { Vacancy as VacancyApi } from '@/modules/vacancy/api';
import { createTestingPinia } from '@pinia/testing';
import { expect, vi } from 'vitest';

vi.mock('@/modules/vacancy/api');
vi.mock('vue3-apexcharts', () => ({
  render: () => '<div>Mocked ApexCharts</div>',
}));

const vacancyList: VacancyTotalProcess[] = [
  {
    schoolYear: {
      year: 2021,
    },
    totalPreRegistrations: 1217,
  },
  {
    schoolYear: {
      year: 2021,
    },
    totalPreRegistrations: 912,
  },
];

const vacancyLoad: VacancyLoadProcesses = {
  grades: [
    {
      id: '64',
      key: '64',
      label: 'Série - 4',
      name: 'Série - 4',
    },
  ],
  periods: [
    {
      id: '1',
      key: '1',
      label: 'Matutino',
      name: 'Matutino',
    },
  ],
  processes: [
    {
      id: '1',
      name: 'TESTE 1',
      unavailable: 0,
      exceded: 0,
      ultrapassed: 0,
      percentual: {
        accepted: 1,
        available: 1,
        unavailable: 1,
        waitingAvailable: 1,
        waitingExceded: 1,
        maxWaiting: 1,
        waitingRealAvailable: 1,
        total: 1,
        totalExceded: 1,
        exceded: 1,
        waiting: 1,
      },
      max: 1,
      totalExceded: 1,
      waitingAvailable: 1,
      waitingExceded: 1,
      maxWaiting: 1,
      process: {
        id: '1',
        name: 'TESTE 1',
      },
      accepted: 1,
      available: 1,
      excededVacancies: 0,
      rejected: 1,
      total: 1,
      waiting: 1,
    },
  ],
  schools: [
    {
      id: '00000',
      key: '00000',
      label: 'ESCOLA ESTADUAL DE ENSINO FUNDAMENTAL',
      name: 'ESCOLA ESTADUAL DE ENSINO FUNDAMENTAL',
    },
  ],
  stats: {
    vacancies: 0,
    total: 0,
    accepted: 0,
    rejected: 0,
    waiting: 0,
  },
  unique: {
    1: 1,
  },
};

const onMounted = async () => {
  vi.mocked(VacancyApi.list).mockResolvedValueOnce(vacancyList);
  vi.mocked(VacancyApi.load).mockResolvedValueOnce(vacancyLoad);

  const wrapper = mount(Vacancies, {
    global: {
      plugins: [createTestingPinia()],
      directives: {
        tooltip: VTooltip,
      },
      stubs: ['x-field'],
    },
  });

  await flushPromises();

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('Vacancies', () => {
  let wrapper: VueWrapper<InstanceType<typeof Vacancies>>;

  beforeEach(async () => {
    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.years).toEqual([
      {
        key: 2021,
        label: 2021,
      },
    ]);
    expect(wrapper.vm.filter).toEqual({
      school: '00000',
      period: null,
      grade: null,
      year: 2021,
      canAcceptInBatch: false,
    });

    expect(wrapper.vm.processes).toEqual(vacancyLoad.processes);
    expect(wrapper.vm.schools).toEqual(vacancyLoad.schools);
    expect(wrapper.vm.grades).toEqual(vacancyLoad.grades);
    expect(wrapper.vm.periods).toEqual(vacancyLoad.periods);
    expect(wrapper.vm.stats).toEqual(vacancyLoad.stats);

    expect(wrapper.vm.charts.length).toBe(1);
    expect(wrapper.vm.charts[0].series).toEqual([1, 1, 1]);
    expect(wrapper.vm.charts[0].id).toEqual('1');
  });

  test('checks the number of cards printed on the screen', () => {
    expect(wrapper.findAll('[data-test="vacancy-card"]').length).toBe(1);
  });

  test('Check if card shown on screen is printing data correctly', () => {
    const card = wrapper.find('[data-test="vacancy-card"]');

    expect(card.find('[data-test="vacancy-card-title"]').text()).toBe(
      'TESTE 1'
    );

    expect(card.find('[data-test="vacancy-card-relation"]').text()).toBe('1/1');
    expect(
      card.find('[data-test="vacancy-card-relation-percentage"]').text()
    ).toBe('100%');
  });

  test('When changing the filters, you must make a new listing of vacancies', async () => {
    vi.mocked(VacancyApi.load).mockResolvedValueOnce(vacancyLoad);

    const filter: VacancyFilter = {
      school: '00000',
      period: null,
      grade: null,
      year: 2021,
      canAcceptInBatch: false,
    };

    await wrapper.setValue(filter, 'filter');

    const btn = wrapper.find('[data-test="vacancy-btn-search"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(wrapper.vm.filter).toEqual(filter);
    expect(wrapper.vm.charts.length).toBe(1);
    expect(wrapper.vm.charts[0].series).toEqual([1, 1, 1]);
    expect(wrapper.vm.charts[0].id).toEqual('1');
  });

  test('When clicking on the "clear" button, the Filter variable must have the original values', async () => {
    vi.mocked(VacancyApi.load).mockResolvedValueOnce(vacancyLoad);

    const btn = wrapper.find('[data-test="vacancy-btn-clear"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(wrapper.vm.filter).toMatchObject({
      school: '00000',
      period: null,
      grade: null,
      year: null,
      canAcceptInBatch: false,
    });

    expect(wrapper.vm.charts.length).toBe(1);
    expect(wrapper.vm.charts[0].series).toEqual([1, 1, 1]);
    expect(wrapper.vm.charts[0].id).toEqual('1');
  });
});
