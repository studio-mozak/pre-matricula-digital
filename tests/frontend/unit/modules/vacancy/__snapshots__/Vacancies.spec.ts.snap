// Vitest Snapshot v1

exports[`Vacancies > should match snapshot 1`] = `
<main>
  <h1>
    Painel de Vagas
  </h1>
  <div
    class="row mt-5 mb-5 pl-lg-0 pl-3 pr-lg-0 pr-3"
  >
    <div
      class="col-12 p-0 d-flex justify-content-lg-around"
      style="overflow-x: auto;"
    >
      
      <div
        class="x-card x-card--bordered col-6 col-md-2 bg-background card-summary mr-3 mr-lg-0"
      >
        <div
          class="x-card__section x-card__section--vert d-flex justify-content-center mt-3 mb-3 pb-0"
        >
          <div
            class="rounded-circle p-3 border-card bg-blue svg-real-estate"
          />
        </div>
        <div
          class="x-card__section x-card__section--vert text-center pt-0 pb-3"
        >
          <p
            class="card-number"
          >
            0
          </p>
          <p
            class="card-text"
          >
            Vagas ofertadas
          </p>
        </div>
      </div>
      <div
        class="x-card x-card--bordered col-6 col-md-2 bg-background card-summary mr-3 mr-lg-0"
      >
        <div
          class="x-card__section x-card__section--vert d-flex justify-content-center mt-3 mb-3 pb-0"
        >
          <div
            class="rounded-circle p-3 border-card bg-blue svg-perfis"
          />
        </div>
        <div
          class="x-card__section x-card__section--vert text-center pt-0 pb-3"
        >
          <p
            class="card-number"
          >
            0
          </p>
          <p
            class="card-text"
          >
            Pré-matrículas inscritas
          </p>
        </div>
      </div>
      <div
        class="x-card x-card--bordered col-6 col-md-2 bg-background card-summary mr-3 mr-lg-0"
      >
        <div
          class="x-card__section x-card__section--vert d-flex justify-content-center mt-3 mb-3 pb-0"
        >
          <div
            class="rounded-circle p-3 border-card bg-green svg-aprovado"
          />
        </div>
        <div
          class="x-card__section x-card__section--vert text-center pt-0 pb-3"
        >
          <p
            class="card-number"
          >
            0
          </p>
          <p
            class="card-text"
          >
            Deferidas
          </p>
        </div>
      </div>
      <div
        class="x-card x-card--bordered col-6 col-md-2 bg-background card-summary mr-3 mr-lg-0"
      >
        <div
          class="x-card__section x-card__section--vert d-flex justify-content-center mt-3 mb-3 pb-0"
        >
          <div
            class="rounded-circle p-3 border-card bg-red svg-desaprovado"
          />
        </div>
        <div
          class="x-card__section x-card__section--vert text-center pt-0 pb-3"
        >
          <p
            class="card-number"
          >
            0
          </p>
          <p
            class="card-text"
          >
            Indeferidas
          </p>
        </div>
      </div>
      <div
        class="x-card x-card--bordered col-6 col-md-2 bg-background card-summary mr-3 mr-lg-0"
      >
        <div
          class="x-card__section x-card__section--vert d-flex justify-content-center mt-3 mb-3 pb-0"
        >
          <div
            class="rounded-circle p-3 border-card bg-yellow svg-esperando"
          />
        </div>
        <div
          class="x-card__section x-card__section--vert text-center pt-0 pb-3"
        >
          <p
            class="card-number"
          >
            0
          </p>
          <p
            class="card-text"
          >
            Em espera
          </p>
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="x-card bg-primary mb-4"
  >
    <div
      class="x-card__section x-card__section--vert pl-2 pr-2"
    >
      <form>
        <div
          class="row m-0 mt-1 filter-gutters"
        >
          <x-field-stub
            allowclear="false"
            bind="[object Object]"
            containerclass="col-12 col-md-2 mb-md-0"
            disabled="false"
            errors="false"
            loading="false"
            modelvalue="2021"
            name="year"
            options="[object Object]"
            placeholder="Ano"
            searchable="false"
            type="SELECT"
          />
          <x-field-stub
            allowclear="false"
            bind="[object Object]"
            containerclass="col-12 col-md-4 mb-md-0"
            disabled="false"
            errors="false"
            loading="false"
            modelvalue="00000"
            name="school"
            options="[object Object]"
            placeholder="Filtrar por escola"
            searchable="true"
            type="SELECT"
          />
          <x-field-stub
            allowclear="false"
            bind="[object Object]"
            containerclass="col-12 col-md mb-md-0"
            disabled="false"
            errors="false"
            loading="false"
            name="grade"
            options="[object Object]"
            placeholder="Filtrar por série"
            searchable="true"
            type="SELECT"
          />
          <x-field-stub
            allowclear="false"
            bind="[object Object]"
            containerclass="col-12 col-md mb-md-0"
            disabled="false"
            errors="false"
            loading="false"
            name="period"
            options="[object Object]"
            placeholder="Filtrar por turno"
            searchable="false"
            type="SELECT"
          />
        </div>
        <div
          class="d-flex m-0 mt-3 filter-gutters justify-content-end"
        >
          <div>
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--flat x-btn--rectangle text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase mr-3"
              data-test="vacancy-btn-clear"
              role="button"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Limpar
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-white border-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase mr-2"
              data-test="vacancy-btn-search"
              role="button"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Buscar
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    class="row legend-wrapper ml-1"
  >
    <div
      class="bg-green mr-1 mb-1"
    />
    <span>
      DEFERIDAS
    </span>
    <div
      class="bg-danger mr-1 mb-1"
    />
    <span>
      INDEFERIDAS
    </span>
    <div
      class="bg-yellow mr-1 mb-1"
    />
    <span>
      EM ESPERA
    </span>
    <div
      class="bg-light mr-1 mb-1"
    />
    <span>
      INDISPONÍVEL
    </span>
    <div
      class="open-legend mr-1 mb-1"
    />
    <span>
      ABERTAS
    </span>
  </div>
  <!--v-if-->
  <div
    class="row"
  >
    
    <div
      class="col-12 col-md-6 mt-4"
      data-test="vacancy-card"
    >
      <div
        class="x-card x-card--bordered"
      >
        <div
          class="x-card__section x-card__section--vert p-3"
        >
          <div
            class="d-flex justify-content-between"
          >
            <h3
              class="font-hind-18-primary text-primary m-0 text-truncate"
              data-test="vacancy-card-title"
            >
              TESTE 1
            </h3>
            <!--v-if-->
          </div>
        </div>
        <div
          class="x-card__section x-card__section--vert"
        >
          <!---->
          <div
            class="row p-3"
          >
            <div
              class="col-12 px-3"
            >
              <div
                class="d-flex"
              >
                <div>
                  <div>
                    Inscrições deferidas
                  </div>
                  <div
                    class="text-muted small"
                  >
                     em relação ao total de 
                    <strong>
                      vagas ofertadas
                    </strong>
                  </div>
                </div>
                <div
                  class="ml-auto text-right"
                >
                  <div
                    data-test="vacancy-card-relation"
                  >
                    1/1
                  </div>
                  <div
                    class="text-muted small"
                    data-test="vacancy-card-relation-percentage"
                  >
                    100% 
                  </div>
                </div>
              </div>
              <div
                class="d-flex mt-2"
                style="margin-left: -0.25rem; margin-right: -0.25rem;"
              >
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-red"
                    style="max-width: 100%; height: 14px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-green"
                    style="max-width: 100%; height: 14px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-light"
                    style="max-width: 100%; height: 14px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    style="max-width: 100%; border: 1px solid #d8e9ff; height: 14px; border-radius: 4px;"
                  />
                </div>
              </div>
              <div
                class="d-flex mt-2"
                style="margin-left: -0.25rem; margin-right: -0.25rem;"
              >
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-danger"
                    style="max-width: 100%; height: 4px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-info"
                    style="max-width: 100%; height: 4px; border-radius: 4px;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="row p-3"
          >
            <div
              class="col-12 px-3"
            >
              <div
                class="d-flex"
              >
                <div>
                  <div>
                    Inscrições em espera
                  </div>
                  <div
                    class="text-muted small"
                  >
                     em relação a quantidade de 
                    <strong>
                      vagas disponíveis
                    </strong>
                  </div>
                </div>
                <div
                  class="ml-auto text-right"
                >
                  <div>
                    1/1
                  </div>
                  <div
                    class="text-muted small"
                  >
                    1% 
                  </div>
                </div>
              </div>
              
              <div
                class="d-flex mt-2"
                style="margin-left: -0.25rem; margin-right: -0.25rem;"
              >
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-yellow"
                    style="max-width: 100%; height: 14px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    style="max-width: 100%; border: 1px solid #d8e9ff; height: 14px; border-radius: 4px;"
                  />
                </div>
              </div>
              <div
                class="d-flex mt-2"
                style="margin-left: -0.25rem; margin-right: -0.25rem;"
              >
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-danger"
                    style="max-width: 100%; height: 4px; border-radius: 4px;"
                  />
                </div>
                <div
                  class="cp"
                  style="width: 1%;"
                >
                  <div
                    class="bg-info"
                    style="max-width: 100%; height: 4px; border-radius: 4px;"
                  />
                </div>
              </div>
              
            </div>
          </div>
          <div
            class="row p-3"
          >
            <div
              class="col-12 px-3"
            >
              <div
                class="d-flex"
              >
                <div>
                  <div>
                    Alunos(as) inscritos(as)
                  </div>
                  <div
                    class="text-muted small"
                  >
                     contabilizados somente os alunos com 
                    <strong>
                      CPF cadastrado
                    </strong>
                  </div>
                </div>
                <div
                  class="ml-auto text-right"
                >
                  <div>
                    1
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="x-card__section x-card__section--vert"
        >
          <div>
            <!--v-if-->
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 mt-3"
              data-test="vacancy-card-see-more"
              role="button"
              style="font-size: 20px;"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Ver mais detalhes
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</main>
`;
